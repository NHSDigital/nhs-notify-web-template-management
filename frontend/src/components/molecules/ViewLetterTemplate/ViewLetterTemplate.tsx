'use client';

import PreviewTemplateDetailsLetter from '@molecules/PreviewTemplateDetails/PreviewTemplateDetailsLetter';
import {
  LetterTemplate,
  PageComponentProps,
} from 'nhs-notify-web-template-management-utils';
import content from '@content/content';
import Link from 'next/link';
import { NHSNotifyMain } from '@atoms/NHSNotifyMain/NHSNotifyMain';
import NotifyBackLink from '@atoms/NHSNotifyBackLink/NHSNotifyBackLink';

export function ViewLetterTemplate({
  initialState,
}: Readonly<PageComponentProps<LetterTemplate> & { owner?: string }>) {
  const { createNewTemplate, backLinkText } =
    content.components.viewSubmittedTemplate;

  return (
    <>
      <Link href='/message-templates' passHref legacyBehavior>
        <NotifyBackLink data-testid='back-to-templates-link'>
          {backLinkText}
        </NotifyBackLink>
      </Link>
      <NHSNotifyMain>
        <div className='nhsuk-grid-row'>
          <div className='nhsuk-grid-column-full'>
            <PreviewTemplateDetailsLetter template={initialState} />
            <p>{createNewTemplate}</p>
            <p>
              <Link
                href='/message-templates'
                data-testid='back-to-templates-link-bottom'
              >
                {backLinkText}
              </Link>
            </p>
          </div>
        </div>
      </NHSNotifyMain>
    </>
  );
}
