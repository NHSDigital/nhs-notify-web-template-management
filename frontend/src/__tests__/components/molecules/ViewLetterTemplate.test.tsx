import { render } from '@testing-library/react';
import { LetterTemplate } from 'nhs-notify-web-template-management-utils';
import { ViewLetterTemplate } from '@molecules/ViewLetterTemplate/ViewLetterTemplate';

describe('ViewLetterTemplate component', () => {
  it('matches submitted snapshot', () => {
    const container = render(
      <ViewLetterTemplate
        initialState={
          {
            templateType: 'LETTER',
            id: 'template-id',
            name: 'Example template',
            templateStatus: 'SUBMITTED',
            createdAt: '2025-03-28T12:30:54.684Z',
            updatedAt: '2025-03-28T12:31:54.684Z',
            letterType: 'x0',
            language: 'en',
            files: {
              pdfTemplate: {
                fileName: 'file.pdf',
                currentVersion: 'a',
                virusScanStatus: 'PASSED',
              },
              proofs: {
                'supplier-proof.pdf': {
                  fileName: 'supplier-proof.pdf',
                  virusScanStatus: 'PASSED',
                  supplier: 'MBA',
                },
              },
            },
          } satisfies LetterTemplate
        }
      />
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
