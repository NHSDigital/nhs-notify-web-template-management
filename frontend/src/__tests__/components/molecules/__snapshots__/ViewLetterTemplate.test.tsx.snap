// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewLetterTemplate component matches submitted snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-back-link"
  >
    <a
      class="nhsuk-back-link__link"
      href="/templates/message-templates"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__chevron-left"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.5 12c0-.3.1-.5.3-.7l5-5c.4-.4 1-.4 1.4 0s.4 1 0 1.4L10.9 12l4.3 4.3c.4.4.4 1 0 1.4s-1 .4-1.4 0l-5-5c-.2-.2-.3-.4-.3-.7z"
        />
      </svg>
      Back to all templates
    </a>
  </div>
  <main
    class="nhsuk-main-wrapper"
    id="maincontent"
    role="main"
  >
    <div
      class="nhsuk-grid-row"
    >
      <div
        class="nhsuk-grid-column-full"
      >
        <h1
          class="preview__heading"
          data-testid="preview-message__heading"
        >
          Example template
        </h1>
        <div
          class="nhsuk-width-container nhsuk-u-margin-bottom-6 nhsuk-body-m"
        >
          <dl
            class="nhsuk-summary-list nhsuk-u-margin-bottom-4 preview"
          >
            <div
              class="nhsuk-summary-list__row"
            >
              <dt
                class="nhsuk-summary-list__key"
              >
                Template ID
              </dt>
              <dd
                class="nhsuk-summary-list__value"
              >
                template-id
              </dd>
            </div>
            <div
              class="nhsuk-summary-list__row"
            >
              <dt
                class="nhsuk-summary-list__key"
              >
                Type
              </dt>
              <dd
                class="nhsuk-summary-list__value"
              >
                Standard letter
              </dd>
            </div>
            <div
              class="nhsuk-summary-list__row"
            >
              <dt
                class="nhsuk-summary-list__key"
              >
                Status
              </dt>
              <dd
                class="nhsuk-summary-list__value"
              >
                <strong
                  class="nhsuk-tag nhsuk-tag--grey"
                >
                  Submitted
                </strong>
              </dd>
            </div>
            <div
              class="nhsuk-summary-list__row"
            >
              <dt
                class="nhsuk-summary-list__key"
              >
                Template file
              </dt>
              <dd
                class="nhsuk-summary-list__value"
              >
                <div
                  class="container"
                >
                  <svg
                    class="icon"
                    fill="none"
                    height="26"
                    viewBox="0 0 20 26"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0V26H20V6.59375L19.7188 6.28125L13.7188 0.28125L13.4062 0H0ZM2 2H12V8H18V24H2V2ZM14 3.4375L16.5625 6H14V3.4375Z"
                      fill="#4C6272"
                    />
                  </svg>
                  <p
                    class="text"
                  >
                    file.pdf
                  </p>
                </div>
              </dd>
            </div>
          </dl>
        </div>
        <p>
          If you want to change a submitted or live template, you must create a new template to replace it.
        </p>
        <p>
          <a
            href="/message-templates"
          >
            Back to all templates
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
